origin-discovery/src/listener/handler_identity.js

```diff
@@ -153,6 +152,52 @@ class IdentityEventHandler {
     }))
   }
 
+  /**
+   * If the user signed up via a referral, links the referrer and referee
+   * by inserting a row in the growth_referral table.
+   * @param {UserModel} user - Origin-js user model object.
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _recordGrowthReferral(user) {
+    if (!user.metadata || !user.metadata.referrerCode) {
+      // Nothing to record, the user did not come from referral program.
+      return
+    }
+
+    const referee = user.address.toLowerCase()
+
+    // Lookup the invite code to get the referrer.
+    const code = await db.GrowthInviteCode.findOne({ where: { code } })
+    if (!code) {
+      logger.error(`Invalid referral code present in identity of ${referee}`)
+      return
+    }
+    const referrer = code.ethAddress
+
+    // Check for any existing referral data for this referee.
+    const row = db.GrowthReferral.findOne({
+      where: {
+        referee_eth_address: referee
+      }
+    })
+    if (row) {
+      if (row.referrerEthAddress != referrer) {
+        // This should never happen.
+        logger.error(`Referee ${referee} already referred by ${row.referrerEthAddress}`)
```

Comment:
Yes, good point. Let me clarify the comment !

Reply:
Hmm... but this bares the question of to which referrer the signup should be attributed to ?
I think it should be attributed to the referrer the user first clicked the link of.
For example:
 - Referrer A sends invite to user X
 - Referrer B also sends invite to friend X
 - User X clicks first on link from referrer A
 - User X later clicks on link from referrer B
 - User X completes the signup flow
=> Referral bonus goes to user A.

So I think the logic should be that we record a single entry for user A in the GrowthReferral table ? I'm actually thinking I should update the schema for the GrowthReferral table to reflect this by adding a unique constraint on referee_eth_address. To prevent more than one entry per referee to get added. Does that makes sense ?

Reply:
Agree with all stated above. It kind of makes sense logically that the first person really introduced you to our product. Does it really matter that the second person referred once you already know of Origin.